nixos/modules/services/web-servers/caddy.nix

```diff
@@ -142,13 +154,13 @@ in {
       };
     };
 
-    users.users.caddy = {
-      group = "caddy";
-      uid = config.ids.uids.caddy;
+    users.users.${cfg.user} = {
```

Comment:
Since you're looking for maximum flexibility I would suggest using the regular pattern in our module system of conditionally provisioning the user and group, only if the default values are provided.